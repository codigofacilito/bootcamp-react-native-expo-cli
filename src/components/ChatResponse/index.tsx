import { FC, memo } from "react";
import { StyleSheet, Text, View } from "react-native";

type ChatResponseProps = {
    response: string;
}

const ChatResponse: FC<ChatResponseProps> = ({ response }) => {
    return (
        <View style={styles.container}>
            <Text style={styles.message}>{response}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    message: {
        fontSize: 14,
    },
    container: {
        paddingVertical: 2,
    },
});

export default memo(ChatResponse);
