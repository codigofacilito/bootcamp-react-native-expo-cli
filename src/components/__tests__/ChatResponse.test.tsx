import { render } from "@testing-library/react-native";

import ChatResponse from "../ChatResponse";

describe("<ChatResponse/> component", () => {
    test("Renderea correctamente dada una respuesta como prop", () => {
        const { getByText } = render(<ChatResponse response="Hola!" />);

        const component = getByText("Hola!");

        expect(component).toBeTruthy();
        expect(component).toBeDefined();

        expect(component.props.style).toEqual({ fontSize: 14 });
    });

    test("Hace match con el snapshoot", () => {
        const { toJSON } = render(<ChatResponse response="Hola Mundo 2s!" />);
        expect(toJSON()).toMatchSnapshot();
    });
});