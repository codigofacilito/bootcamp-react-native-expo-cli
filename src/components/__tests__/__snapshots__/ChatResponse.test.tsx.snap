// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatResponse/> component Hace match con el snapshoot 1`] = `
<View
  style={
    {
      "paddingVertical": 2,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 14,
      }
    }
  >
    Hola Mundo 2s!
  </Text>
</View>
`;
